### PR TITLE
feat: toggle data view

### PR DESCRIPTION
**Pull request by niekvanstaveren**
## Motivation

Adds support for toggleDataView. Needed to provide an accessible way to the data in Nebula mashups. 

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [x] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/viz.js [hunk 1]
```diff
@@ -149,6 +150,51 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
       const propertyTree = await conversionConvertTo({ halo, model, cellRef, newType });
       return propertyTree;
     },
+    /**
+     * Toggles the view of the chart into an accessible table
```
Comment (veinfors) [inline apis/nucleus/src/viz.js:154]: Maybe explain that the type you want to toggle to, sn-table by default, needs to be registered on the embed instance to be able to do this?

Comment (niekvanstaveren) [reply apis/nucleus/src/viz.js]: Added some more info and also a check if the type is actually registered

File: apis/nucleus/src/viz.js [hunk 2]
```diff
@@ -149,6 +150,44 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
       const propertyTree = await conversionConvertTo({ halo, model, cellRef, newType });
       return propertyTree;
     },
+    /**
+     * Toggles the view of the chart into an accessible table. The type (default sn-table) needs to be registered in order to use this functionality
+     *
+     * @since 4.6.0
```
Comment (niekvanstaveren) [inline apis/nucleus/src/viz.js:156]: Should it maybe start as experimental?

File: apis/nucleus/src/index.js [hunk 3]
```diff
@@ -547,6 +547,15 @@ function nuked(configuration = {}) {
        * //]
        */
       getRegisteredTypes: types.getList,
+      /**
+       * Check if certain visualization type is registered
+       * @function
+       * @param {string=} type Type to check
```
Comment (veinfors) [inline apis/nucleus/src/index.js:553]:
```suggestion
       * @param {string} type Type to check
```

File: apis/nucleus/src/viz.js [hunk 4]
```diff
@@ -149,6 +150,44 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
...
+     * @param {string} newType - The type used for the view data table
```
Comment (veinfors) [inline apis/nucleus/src/viz.js:157]:
```suggestion
     * @param {string=} [newType=sn-table] The type used for the view data table
```

File: apis/nucleus/src/viz.js (hunk 2 above)
Comment (veinfors) [reply apis/nucleus/src/viz.js:156]: Sure, we could start by having it experimental...

**Review by veinfors — APPROVED** [review]

**Comment by Caele** [comment]
Some notes:
- Need to decide on the API itself: viz.toggleDataView(type)  has a few options to it how it should look, and hard to change it later. It is tied to conversion so might make sense to place it there viz.convert.toggleDataView . Also is a question if it really should be a toggle or not.
- The created session objects are currently not reused nor destroyed, so new ones are constantly added.

**Pull request merged**